### ISSUE TITLE
A change to an editor tab shouldn't change its instance

### ISSUE THREAD

**Issue by jrieken** [microsoft/vscode#144664]
The extension host side of editor tabs currently creates new instances of tab whenever something changes. E.g tab instance is thrown away and recreated when it dirty state changes. This behaviour makes it hard to work with the API and instances should update in-place whenever possible.

**Comment by lramos15** [comment]
Yeah I understand this is probably the perfect solution but I didn't want to have to patch the tab model in the ext host and main thread due to the complexity of it. All the cases I can think of or have seen would either use the instance returned from an event or just will grab the model when a command is executed

**Comment by jrieken** [comment]
👯
